src/lib/optimizer/optimizer.cpp

```diff
@@ -101,6 +102,8 @@ std::shared_ptr<Optimizer> Optimizer::create_default_optimizer() {
 
   optimizer->add_rule(std::make_shared<JoinOrderingRule>(std::make_shared<CostModelLogical>()));
 
+  optimizer->add_rule(std::make_shared<LikeReplacementRule>());
```

Comment:
Is there a special reason why this rule will be applied after the `JoinOrderingRule`? I am not sure, if there is a preferred order (as long as it works). But it seems that in general, rules that replace nodes are executed before rules that change the order of nodes?